src/libraries/System.Private.CoreLib/src/System/Runtime/CompilerServices/RequiresLocationAttribute.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.ComponentModel;
+
+namespace System.Runtime.CompilerServices
+{
+    /// <summary>
+    /// Reserved for use by a compiler for tracking metadata.
+    /// This attribute should not be used by developers in source code.
+    /// </summary>
+    [EditorBrowsable(EditorBrowsableState.Never)]
+    [AttributeUsage(AttributeTargets.Parameter, Inherited = false)]
+    public sealed class RequiresLocationAttribute : Attribute
+    {
+        /// <summary>Initializes the attribute.</summary>
+        public RequiresLocationAttribute()
```

Comment:
Just curious — how come so many of these attributes in the BCL have a parameterless constructor at all? As in why are they not just empty and relying on the default constructor? 🤔

Reply:
Either way we get the same ctor, and this gives us a place to put the XML comments, which then seed the docs. It also prevents additional churn in the future if we add an additional ctor, which would then prevent the compiler from injecting the parameterless one.

Reply:
Oh, I see, hadn't considered XML docs, that makes sense. Thank you! 😄